library/core/src/num/nonzero.rs

```diff
@@ -895,3 +895,39 @@ macro_rules! nonzero_unsigned_is_power_of_two {
 }
 
 nonzero_unsigned_is_power_of_two! { NonZeroU8 NonZeroU16 NonZeroU32 NonZeroU64 NonZeroU128 NonZeroUsize }
+
+macro_rules! nonzero_min_max {
+    ( $( $MinVal:expr , $Ty: ident($Int: ty); )+ ) => {
+        $(
+            impl $Ty {
+                #[unstable(feature = "nonzero_max", issue = "89065")]
+                #[doc = concat!("The maximum value for a`", stringify!($Ty), "` is the same as `", stringify!($Int), "`")]
+                #[doc = concat!("assert_eq!(", stringify!($Ty), "::MAX, ", stringify!($Int), "::MAX);")]
+                // SAFETY: Since the MAX value, for any supported integer type, is greater than 0, the MAX will always be non-zero.
+                pub const MAX : $Ty = unsafe { $Ty::new_unchecked(<$Int>::MAX) };
+                #[unstable(feature = "nonzero_min", issue = "89065")]
+                #[doc = concat!("The minimum value for a`", stringify!($Ty), "`.")]
+                /// # Examples
+                #[doc = concat!("assert_eq!(", stringify!($Ty), "::MIN, ", stringify!($MinVal), ";")]
+                // SAFETY: In the signed case, the minimum integer is negative, and therefore non-zero.
+                // SAFETY:  In the unsignedd case, we use one, which is non-zero.
+                pub const MIN : $Ty = unsafe { $Ty::new_unchecked($MinVal)};
+                }
```

Comment:
Indentation looks off here. Did the formatter really demand it like that?

Reply:
It's a macro, it should be formatted manually.

Reply:
Do you have specific advice on how to format it? I noticed that macros needed manual formatting. Wasn't sure if I had it right though @ibraheemdev .

Reply:
This closing brace, and the analogous one in `nonzero_constants_signed` should be unindented by one level = four spaces. The rest looks fine.